go/host/p2p/p2p.go

```diff
@@ -131,36 +134,37 @@ func (p *p2pImpl) SendBatches(batchMsg *host.BatchMsg, to string) error {
 	return p.send(msg, to)
 }
 
-// Status returns the current status of the lib
+// Status returns the current status of the p2p layer
 func (p *p2pImpl) Status() *host.P2PStatus {
-	return p.status.status()
+	return p.metrics.status()
 }
 
 // HealthCheck returns whether the p2p is considered healthy
 // Currently it considers itself unhealthy
 // if there's more than 100 failures on a given fail type
 // if there's a known peer for which a message hasn't been received
 func (p *p2pImpl) HealthCheck() bool {
-	const thresholdPerFailure = 100
-	currentStatus := p.status.status()
+	currentStatus := p.metrics.status()
 
-	if sumFailures(currentStatus.FailedMessageDecodes) >= thresholdPerFailure ||
-		sumFailures(currentStatus.FailedMessageReads) >= thresholdPerFailure ||
-		sumFailures(currentStatus.FailedSendMessagesPeerConnection) >= thresholdPerFailure ||
-		sumFailures(currentStatus.FailedSendMessageWrites) >= thresholdPerFailure {
+	if currentStatus.FailedReceivedMessages >= _thresholdErrorFailure ||
+		currentStatus.FailedSendMessage >= _thresholdErrorFailure {
 		return false
 	}
 
-	if time.Now().After(p.peerAddressesLastUpdate.Add(_peerListCoolOffPeriod)) {
-		if peerAddrs := peerNoMessage(currentStatus.ReceivedMessages, p.peerAddresses); len(peerAddrs) != 0 {
-			p.logger.Warn("not received any message from peer(s) in the alert period",
+	var noMsgReceivedPeers []string
+	for peer, lastMsgTimestamp := range p.peerTracker.receivedMessagesByPeer() {
+		if time.Now().After(lastMsgTimestamp.Add(_alertPeriod)) {
+			noMsgReceivedPeers = append(noMsgReceivedPeers, peer)
+			p.logger.Warn("no message from peer in the alert period",
 				log.HostCmp, p.ourAddress,
-				"peers", peerAddrs,
-				"alertPeriod", _rollingPeriod,
+				"peer", peer,
+				"alertPeriod", _alertPeriod,
 			)
-			return false
 		}
 	}
+	if len(noMsgReceivedPeers) > 0 { //nolint: gosimple
+		return false
```

Comment:
nice
this will report a node as unhealthy if it's not receiving any messages from anyone, right?

Reply:
umm. But that makes the peer unhealthy.
This current node is fine